[cutlass backend] Cache config generation locally and remotely

henrylhtsang opened the pull request:
Summary:
Trying to cache the json list of configs.

There are probably some more work:
* preset
* filelock (?)
* for cases where we generate from scratch, save it to local as well (?)

Test Plan: tested offline

Reviewed By: coconutruben

Differential Revision: D75334439




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

ColinPeppler commented on torch/_inductor/codegen/cuda/cutlass_cache.py:
```diff
@@ -0,0 +1,93 @@
+
+@clear_on_fresh_inductor_cache
+@functools.lru_cache(None)
+def fetch_ops() -> Optional[dict[Any, Any]]:
```
nit: rename to maybe_fetch_ops

henrylhtsang committed:
[cutlass backend] Cache config generation locally and remotely (#154686)

Summary:

Trying to cache the json list of configs.

There are probably some more work:
* preset
* filelock (?)
* for cases where we generate from scratch, save it to local as well (?)

Test Plan: tested offline

Reviewed By: coconutruben

Differential Revision: D75334439
